### ISSUE TITLE
is_hydrated bug in version 0.4.6

### ISSUE THREAD

**Issue by PabloAsensio** [reflex-dev/reflex#2989]
I've just update to the newest version (0.4.6) and now I'm having this error:

In safari:
```
Unhandled Runtime Error
TypeError: dispatch1["state"] is not a function. (In 'dispatch1["state"]({
                is_hydrated: false
            })', 'dispatch1["state"]' is undefined)
Call Stack
map
[native code]
```

<img width="953" alt="Screenshot 2024-04-02 at 20 58 31" src="https://github.com/reflex-dev/reflex/assets/44872361/0d7d8b96-c671-4042-b890-b42de9ac7cab">

In chrome:

![Screenshot 2024-04-02 at 21 06 35](https://github.com/reflex-dev/reflex/assets/44872361/7d1f0eb6-7e2d-497d-9829-a081f4a69129)

 - Python Version: 3.12.1
 - Reflex Version: 0.4.6
 - OS: MacOS Latest
 - Browser (Optional): Safari/Chrome

If I downgrade to 0.4.5 everything is ok.

@masenf  I think the error comes from your #2949 PR.

**Comment by PasqualePuzio** [comment]
Just wondering: does your app have a state or it's stateless?

**Comment by picklelo** [comment]
@PabloAsensio thanks for reporting - we have a fix for this that will merge by the next release.

**Comment by azidancorp** [comment]
I don't know if its related, but I created a brand new project following the instructions, and it happens on both the blank and the sidebar template.

```
python3 -m venv .venv
source .venv/bin/activate
reflex init
reflex run
```

```
Unhandled Runtime Error
Error: Hydration failed because the initial UI does not match what was rendered on the server.

Warning: Expected server HTML to contain a matching <div> in <div>.

See more info here: https://nextjs.org/docs/messages/react-hydration-error

Component Stack
div
eval
eval
$f631663db3294ace$export$c760c09fdd558351
Provider
$a093c7e1ec25a057$export$f78649fb9ca566b8
eval
eval
RadixThemesColorModeProvider
AppWrap
f
$
MyApp
Call Stack
throwOnHydrationMismatch
node_modules/react-dom/cjs/react-dom.development.js (12507:0)
tryToClaimNextHydratableInstance
node_modules/react-dom/cjs/react-dom.development.js (12535:0)
updateHostComponent
node_modules/react-dom/cjs/react-dom.development.js (19902:0)
beginWork
node_modules/react-dom/cjs/react-dom.development.js (21618:0)
HTMLUnknownElement.callCallback
node_modules/react-dom/cjs/react-dom.development.js (4164:0)
Object.invokeGuardedCallbackDev
node_modules/react-dom/cjs/react-dom.development.js (4213:0)
invokeGuardedCallback
node_modules/react-dom/cjs/react-dom.development.js (4277:0)
beginWork$1
node_modules/react-dom/cjs/react-dom.development.js (27451:0)
performUnitOfWork
node_modules/react-dom/cjs/react-dom.development.js (26557:0)
workLoopSync
node_modules/react-dom/cjs/react-dom.development.js (26466:0)
renderRootSync
node_modules/react-dom/cjs/react-dom.development.js (26434:0)
performConcurrentWorkOnRoot
node_modules/react-dom/cjs/react-dom.development.js (25738:0)
workLoop
node_modules/scheduler/cjs/scheduler.development.js (266:0)
flushWork
node_modules/scheduler/cjs/scheduler.development.js (239:0)
MessagePort.performWorkUntilDeadline
node_modules/scheduler/cjs/scheduler.development.js (533:0)

```

**Comment by masenf** [comment]
Should be fixed in 0.4.7

**Comment by PabloAsensio** [comment]
> Should be fixed in 0.4.7

Apparently it is :)
